### PR TITLE
Adding the product link to the sms

### PR DESCRIPTION
**Pull request by codonnell822**
did a small refactor just to get my feet wet too. first real time messing with python

Just used requests lib cause I couldn't totally figure out how to use this library: https://developers.google.com/api-client-library/python/apis/urlshortener/v1.

Also, some other questions for you:

- how do you intelligently set those environment variables for your project (locally that is)? do you just keep them in your bash profile or do you dynamically set them somehow just for the project?
- how do you test the worker? web?  do you just do `python3 web.py` and `python3 worker.py` ?

### CODE REVIEW

File: worker.py [hunk 1]
```diff
@@ -20,6 +22,14 @@
     password=os.environ['REDIS_PASS'])
 
 
+def url(link):
+    url = 'https://www.googleapis.com/urlshortener/v1/url?key=%s' % GOOGL_API_KEY
```
Comment (aj-may) [inline worker.py]: To be consistent lets use .format() throughout.

File: worker.py [hunk 2]
```diff
@@ -20,6 +22,14 @@
...
+    headers = {'Content-Type': 'application/json'}
+    payload = json.dumps({'longUrl': link})
+    response = requests.post(url, data=payload, headers=headers)
```
Comment (aj-may) [inline worker.py]: the requests lib will take a dict.  No need to do a json.dumps first.

File: worker.py [hunk 3]
```diff
@@ -20,6 +22,14 @@
...
+    data = json.loads(response.text)
```
Comment (aj-may) [inline worker.py]: the response object has a json() method that will parse the json for you.  json.loads is not necessary.

File: worker.py [hunk 4]
```diff
@@ -7,11 +7,13 @@
 import requests
 import schedule
 import time
+import json
```
Comment (aj-may) [inline worker.py]: requests lib can take care of the json stuffs.  we can drop this.

File: web.py [hunk 5]
```diff
@@ -50,21 +50,23 @@ def subscribe():
 
 @app.route("/message/", methods=['POST'])
 def read_message():
-    if request.form['Body'].lower().strip() == "meh":
+    reply_txt = request.form['Body'].lower().strip()
```
Comment (aj-may) [inline web.py]: can we call this `action`

File: worker.py (hunk 1 above)
Comment (codonnell822) [inline worker.py]: removed this...figured out the library. check out the changes again

File: worker.py (hunk 2 above)
Comment (codonnell822) [inline worker.py]: ok, maybe that would work, I was having trouble getting the google api to accept the request at first, cause I wasn't specifying a content type. I just had

```py
requests.post(url, data={'longUrl': link})
```

and it wasn't matching up with what google was expecting.

moot now, code is changed, but curious to know the best/cleanest way to send that request

File: worker.py (hunk 3 above)
Comment (codonnell822) [inline worker.py]: so just:

```py
data = response.json()
```

?

File: web.py (hunk 5 above)
Comment (codonnell822) [inline web.py]: changed, I struggled with naming this, wasn't sure

**Comment by codonnell822** [comment]
just need to create a new project here: https://console.developers.google.com/project and enable the URL Shortener api, generate a key, blah blah

File: requirements.txt [hunk 6]
```diff
@@ -14,3 +14,4 @@ requests==2.7.0
 schedule==0.3.1
 six==1.9.0
 twilio==4.2.0
+google-api-python-client
```
Comment (aj-may) [inline requirements.txt]: Let's lock this to a version.

File: requirements.txt [hunk 7]
```diff
@@ -14,3 +14,4 @@ requests==2.7.0
 schedule==0.3.1
 six==1.9.0
 twilio==4.2.0
+google-api-python-client=1.4.0
```
Comment (aj-may) [inline requirements.txt]: s/=/==

**Pull request merged**